### PR TITLE
[@mantine/core][Pagination] Add a feature to jump to the start/end of a pagination.

### PR DESCRIPTION
**Pull request by Ultra-Instinct-05**
This PR adds the ability to jump to the start & end of a pagination list.

This feature is optional and controlled by adding the props
1. `withEdges`: Whether to render the buttons that allows a user to jump to start/end page in a pagination. 

### Checkists
- [x] Storybook demos.
- [x] Tests
- [x] Relevant Documentation (Will be completed once the prop names are finalized)

### CODE REVIEW

File: src/mantine-core/src/components/Pagination/DefaultItem/DefaultItem.tsx [hunk 1]
```diff
@@ -2,9 +2,11 @@ import React from 'react';
 import { PrevIcon } from '../icons/PrevIcon';
 import { NextIcon } from '../icons/NextIcon';
 import { DotsIcon } from '../icons/DotsIcon';
+import { ShowFirst } from '../icons/ShowFirst';
+import { ShowLast } from '../icons/ShowLast';
 
 export interface PaginationItemProps extends React.ComponentPropsWithoutRef<'button'> {
-  page: number | 'dots' | 'prev' | 'next';
+  page: number | 'dots' | 'prev' | 'next' | 'showFirst' | 'showLast';
```
Comment (rtivital) [inline src/mantine-core/src/components/Pagination/DefaultItem/DefaultItem.tsx:9]: Let's rename:
- showFirst -> first
- showLast -> last

File: src/mantine-core/src/components/Pagination/Pagination.story.tsx [hunk 2]
```diff
@@ -16,4 +16,11 @@ storiesOf('@mantine/core/Pagination', module)
   .add('With 2 boundaries', () => <Pagination total={20} boundaries={2} initialPage={12} />)
   .add('With 2 boundaries and 2 siblings', () => (
     <Pagination siblings={2} boundaries={2} total={20} initialPage={12} />
+  ))
+  .add('With jump to first/last buttons', () => (
+      <div style={{ padding: 40 }}>
+        <Pagination total={20} boundaries={2} initialPage={12} showFirstButton />
```
Comment (rtivital) [inline src/mantine-core/src/components/Pagination/Pagination.story.tsx:22]: Let's combine showFirstButton and showLastButton props into one and name it `withEdges`

**Review by rtivital — COMMENTED** [review]
Hi @Ultra-Instinct-05, thanks for PR, everything seems fine, just some props renames, please make sure that your branch is in sync with `emotion` branch and has no conflicts with it

File: src/mantine-core/src/components/Pagination/Pagination.story.tsx (hunk 2 above)
Comment (Ultra-Instinct-05) [reply src/mantine-core/src/components/Pagination/Pagination.story.tsx:22]: I thought the above approach allows a more fine grained control over whether a user wants to render single or both the icons ? With `withEdges` both get rendered at once, right ?

Comment (rtivital) [reply src/mantine-core/src/components/Pagination/Pagination.story.tsx:22]: Yes, it would be easier like this

**Comment by Ultra-Instinct-05** [comment]
@rtivital I have made said changes and (hoping I have not made a mess) resolved any conflicts.

**Comment by rtivital** [comment]
Alright, thanks, everything is fine, great job! Please add a new demo to documentation and I'll merge it

**Comment by rtivital** [comment]
Alright, everything is fine, thanks for PR, great job!

**Pull request merged**